modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/file_info_query.rb

```diff
@@ -34,58 +34,80 @@ module StorageInteraction
       module OneDrive
         class FileInfoQuery
           FIELDS = %w[id name fileSystemInfo file folder size createdBy lastModifiedBy parentReference].freeze
-          Auth = ::Storages::Peripherals::StorageInteraction::Authentication
 
           def self.call(storage:, auth_strategy:, file_id:)
             new(storage).call(auth_strategy:, file_id:)
           end
 
           def initialize(storage)
             @storage = storage
-            @delegate = Internal::DriveItemQuery.new(storage)
+            @drive_item_query = Internal::DriveItemQuery.new(storage)
+            @error_data = StorageErrorData.new(source: self.class)
           end
 
           def call(auth_strategy:, file_id:)
             if file_id.nil?
               return ServiceResult.failure(
                 result: :error,
-                errors: ::Storages::StorageError.new(code: :error,
-                                                     data: StorageErrorData.new(source: self.class),
-                                                     log_message: "File ID can not be nil")
+                errors: StorageError.new(code: :error,
+                                         data: @error_data, log_message: "File ID can not be nil")
               )
             end
 
-            Auth[auth_strategy].call(storage: @storage) do |http|
-              @delegate.call(http:, drive_item_id: file_id, fields: FIELDS).map(&storage_file_infos)
+            user_bound_result = Authentication[auth_strategy].call(storage: @storage) do |http|
```

Comment:
🟡 shouldn't we differentiate here? right now you assume, that the first result is a `user_bound_result`. But one can directly call the `file_info_query` with a user-less strategy. I'd rather rename it to something like `file_info_result`. And later in the check combine the conditions `if sr.result == :not_found && auth_strategy.user.present?`.

Reply:
Makes sense, I'll make the change